pkg/web/client.go

```diff
@@ -65,12 +69,47 @@ func NewClient(dockerCommand command.Command, client *http.Client) *Client {
 	}
 }
 
-func (c *Client) PostNewVersion(ctx context.Context, image string, weights []File, files []File) error {
+func (c *Client) PostBuildStart(ctx context.Context, imageHash string, buildTime time.Duration) error {
+	jsonBody := map[string]any{
+		"image_hash":      imageHash,
+		"build_time":      types.Duration(buildTime).String(),
+		"push_start_time": time.Now().UTC(),
+	}
+
+	jsonData, err := json.Marshal(jsonBody)
+	if err != nil {
+		return util.WrapError(err, "failed to marshal JSON for build start")
+	}
+
+	url := webBaseURL()
+	url.Path = strings.Join([]string{"", "api", "models", "build-start"}, "/")
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, url.String(), bytes.NewReader(jsonData))
+	if err != nil {
+		return err
+	}
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusCreated {
+		return errors.New("Bad response from new version endpoint: " + strconv.Itoa(resp.StatusCode))
```

Comment:
nit: it's not a huge deal in this context since the process is not long-lived, but `errors.New` should generally not be used like this. Instead, define a module-level `var` of the base error and then use error wrapping to include the variable bit.